metrics/f1/f1.py

```diff
@@ -96,9 +104,24 @@
 """
 
 
+@dataclass
+class F1Config(Config):
+
+    name: str = "default"
+
+    pos_label: Union[str, int] = 1
+    average: str = "binary"
+    labels: List[str] = None
+    sample_weight: List[float] = None
+
+
 @evaluate.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
 class F1(evaluate.Metric):
-    def _info(self):
+
+    BUILDER_CLASS = F1Config()
```

Comment:
Sorry I meant CONFIG_CLASS. And it doesn't have to be instantiated.

```suggestion
    CONFIG_CLASS = F1Config
```

This way you don't carry the same config for all instances in `self.BUILDER_CLASS`. And instead of
```python
        self.BUILDER_CLASS.update(kwargs)
        info = self._info(self.BUILDER_CLASS)
```
you can do
```python
        info = self._info(self.CONFIG_CLASS(**kwargs))
```